mathics/builtin/box/boxexpression.py

```diff
@@ -0,0 +1,334 @@
+# cython: language_level=3
+# -*- coding: utf-8 -*-
+
+import typing
+from typing import Any, Optional
+
+from mathics.core.element import ensure_context
+
+from mathics.core.symbols import (
+    BaseElement,
+    Symbol,
+)
+
+# from mathics.core.systemsymbols import (
+#    SymbolSymbol,
+# )
+
+
+class BoxError(Exception):
+    def __init__(self, box, form) -> None:
+        super().__init__("Box %s cannot be formatted as %s" % (box, form))
+        self.box = box
+        self.form = form
+
+
+class BoxExpression(BaseElement):
+    head: "Element"
+    leaves: typing.List[Any]
+    _sequences: Any
+
+    def __init__(self, *args, **kwargs):
+        if len(args) == 0:
+            return
+        head = args[0]
+        if isinstance(head, str):
+            head = Symbol(head)
+        self._head = head
+        self._elements = args[1:]
+
+    def __repr__(self) -> str:
+        return "<BoxExpression: %s>" % self
+
+    def __str__(self) -> str:
+        if not hasattr(self, "_head"):
+            return "<BoxExpression uninitialized>"
+        return "%s[%s]" % (
+            self._head,
+            ", ".join([element.__str__() for element in self._elements]),
+        )
+
+    @property
+    def elements(self):
+        return self._elements
+
+    def evaluate(self, evaluation):
+        print("Deprecated!")
+        return self
+
+    def flatten_pattern_sequence(self, evaluation):
+        """
+        called from mathics.core.pattern
+        """
+        print("Deprecated!")
```

Comment:
With this we can't see what is deprecated just reading the logs, maybe it could print "flatten_pattern_sequence is deprecated".

But this is WIP, so I probably shouldn't write this comment.

Reply:
Yes @TiagoCavalcante, I just put this to debug the code.